client/client.go

```diff
@@ -923,16 +926,28 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 	}
 
 	var (
-		pubKey    data.PublicKey
-		err       error
-		errFmtMsg string
+		pubKey     data.PublicKey
+		pubKeyList data.KeyList
+		err        error
+		errFmtMsg  string
 	)
-	switch serverManagesKey {
-	case true:
+
+	switch {
+	case serverManagesKey:
 		pubKey, err = rotateRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
+		pubKeyList = make(data.KeyList, 0, 1)
+		pubKeyList = append(pubKeyList, pubKey)
 		errFmtMsg = "unable to rotate remote key: %s"
-	default:
+	case !serverManagesKey && len(keyList) > 0:
+		pubKeyList = make(data.KeyList, 0, len(keyList))
+		for _, keyID := range keyList {
+			pubKey = r.CryptoService.GetKey(keyID)
+			pubKeyList = append(pubKeyList, pubKey)
+		}
```

Comment:
@HuKeping That's true - it won't cause any panics.  But then an empty list will be passed to `rootFileChangeList`, which might eventually result in either a failure in signing or a rejected update because you can't have an empty list of keys.  It's probably a more friendly error if we just say right away that there were invalid keys, as @ecordell has done here.  :)